packages/components/src/components/drawer/drawer.jsx

```diff
@@ -0,0 +1,87 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+class Drawer extends React.PureComponent {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            is_open: true,
+        };
+    }
+
+    toggleDrawer = () => {
+        this.setState({ is_open: !this.state.is_open });
+        if (this.props.toggleDrawer) {
+            this.props.toggleDrawer(this.state.is_open);
+        }
+    };
+
+    render() {
+        const {
+            children,
+            className,
+            footer,
+            header,
+        } = this.props;
+
+        return (
+            <div className={classNames('dc-drawer',
+                this.state.is_open && `dc-drawer--${'open'}`)}
+            >
+                <div className={classNames('dc-drawer__container',
+                    className && `dc-drawer__container--${className}`)}
+                >
+                    {header &&
+                        <div className={classNames('dc-drawer__header',
+                            className && `dc-drawer__header--${className}`)}
```

Comment:
I think we don't want to pass the same className to container, header, content, footer?

Reply:
@oskar-binary this way we allow ppl to change styles from their apps. for instance if they want to change the style for header they do 
`.dc-drawer__header {
        &--bot-drawer {
            min-width: 28%;
        }
 `
we have to follow this anywhere we want to change override a style of a component

Reply:
Yes but what if bot wants a different className for header and footer?

Reply:
@oskar-binary when you pass a classname we can add it to all right? it will be `drawer__header--bot` and `drawer__footer--bot`. 
better than sending 3 classnames as props?! and user set style for each he needs and the rest will be not used.